go/vt/mysqlctl/mysqld.go

```diff
@@ -157,18 +152,16 @@ func (mysqld *Mysqld) RunMysqlUpgrade() error {
 		return nil
 	}
 
-	// run the program, if it fails, we fail
+	// Run the program, if it fails, we fail.  Note in this
+	// moment, mysqld is running with no grant tables on the local
+	// socket only, so this doesn't need any password for root.
 	args := []string{
 		// --defaults-file=* must be the first arg.
 		"--defaults-file=" + mysqld.config.path,
 		"--socket", mysqld.config.SocketFile,
-		"--user", mysqld.dba.Uname,
+		"--user", "root",
```

Comment:
So you decided to not use "dummy_user"? Why? Could be a nice way to document that authentication doesn't really work at this point.